tests/test_remove_duplicate.py

```diff
@@ -48,3 +79,13 @@ def test():
     for ihalo in range(2, 10):
         halo = tangos.get_halo(ihalo)
         assert halo["Mvir"] == ihalo
+
+    # Now halo 1 should have one less link, which are all unique
```

Comment:
These are good tests. I think it would be ideal to add a couple more

* testing whether two identical links with different names survive (see above comment about `relation_id`)
* testing whether multiple links from the same halo and with the same name and weight, but linking to different halos survive
* if you also implement the idea about deleting the non-maximal duplicates where the weights differ, that could also be tested here

Reply:
Thanks I will implement these additional tests.